src/filters/GeltungsbereicheMenuFilter/GeltungsbereicheMenuFilter.js

```diff
@@ -0,0 +1,167 @@
+
+const useStyles = makeStyles(() => ({
+  selectWrapper: {
+    position: 'relative',
+  },
+  paper: {
+    overflow: 'hidden auto',
+    maxHeight: 350,
+    width: 268,
+  },
+  item: {
+    display: 'block',
+    textOverflow: 'ellipsis',
+  },
+  input: {
+    width: 270,
+  },
+  value: {
+    maxWidth: 190,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  clearBtn: {
+    position: 'absolute',
+    top: 14,
+    right: (props) => (props.isMobile ? 40 : 30),
+  },
+}));
+
+const propTypes = {
+  topic: PropTypes.shape().isRequired,
+};
+
+const GeltungsbereicheMenuFilter = ({ topic }) => {
+  const { t, i18n } = useTranslation();
+  const nullOption = {
+    value: 'gb-null',
+    label: 'Alle Produkte',
+  };
+  const [filterValue, setFilterValue] = useState(nullOption);
+  const screenWidth = useSelector((state) => state.app.screenWidth);
+  const isMobile = useMemo(() => {
+    return ['xs'].includes(screenWidth);
+  }, [screenWidth]);
+  const classes = useStyles({ isMobile });
+
+  const providerChoices = useMemo(
+    () => [
+      nullOption,
+      ...Object.keys(geltungsbereicheMapping).map((key) => {
+        return {
+          value: key,
+          label: geltungsbereicheMapping[key],
+        };
+      }),
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [i18n.language],
+  );
+
+  const layers = topic.layers.filter((l) => {
+    return /^ch.sbb.geltungsbereiche-/.test(l.key);
+  });
+
+  const onChange = (value) => {
+    layers.flat().forEach((layer) => {
+      if (layer) {
+        const { mbMap } = layer.mapboxLayer;
+        const style = mbMap.getStyle();
+        if (!mbMap || !style) {
+          return;
+        }
+        const styleLayer = style.layers.find((l) => {
+          const regex = new RegExp(
+            `^geltungsbereiche.${layer.key.split('-')[1]}$`,
+            'gi',
+          );
+          return regex.test(l.id);
+        });
+        const newStyleLayer = { ...styleLayer };
+        // Remove previous product filters.
+        newStyleLayer.filter = newStyleLayer.filter.filter((fil) => {
+          if (
+            Array.isArray(fil) &&
+            Array.isArray(fil[2]) &&
+            fil[2][1] === 'geltungsbereiche_str'
+          ) {
+            return false;
+          }
+          return true;
+        });
+
+        if (value && value !== nullOption.value) {
+          newStyleLayer.filter.push([
+            'in',
+            value,
+            ['get', 'geltungsbereiche_str'],
+          ]);
+        }
+        // eslint-disable-next-line no-param-reassign
+        layer.styleLayers = [newStyleLayer];
+        layer.removeStyleLayers();
+        layer.addStyleLayers();
+      }
+    });
+    setFilterValue(providerChoices.find((opt) => opt.value === value));
+  };
+
+  return (
+    <div className={`wkp-topic-filter ${classes.selectWrapper}`}>
+      <InputLabel shrink htmlFor="gb-product-filter">
```

Comment:
I'm not sure but shouldn't we use a FromControl around the Input and selectBox ?